### PR TITLE
[libc] Add backup definition for LONG_WIDTH in limits-macros.h.

### PR DESCRIPTION
**Pull request by lntue**

### CODE REVIEW

**Review by SchrodingerZhu — APPROVED** [review]
Lgtm

File: libc/include/llvm-libc-macros/limits-macros.h [hunk 1]
```diff
@@ -69,8 +69,8 @@
 #define LONG_WIDTH __LONG_WIDTH__
 #elif defined(__WORDSIZE)
 #define LONG_WIDTH __WORDSIZE
-#else
-#error "Unknown WORDSIZE to define LONG_WIDTH."
+#else // use sizeof(long) * CHAR_BIT as backup.
+#define LONG_WIDTH (sizeof(long) * CHAR_BIT)
```
Comment (nickdesaulniers) [inline libc/include/llvm-libc-macros/limits-macros.h:73]: Consider adding a comment that this is necessary for clang-11 (and any other clang versions; use godbolt.org to test quickly)

Comment (nickdesaulniers) [reply libc/include/llvm-libc-macros/limits-macros.h:73]: clang-14 is the first release to define `__LONG_WIDTH__`.

**Review by nickdesaulniers — APPROVED** [review]

File: libc/include/llvm-libc-macros/limits-macros.h (hunk 1 above)
Comment (nickdesaulniers) [reply libc/include/llvm-libc-macros/limits-macros.h:74]: ALSO RIP platforms where CHAR_BIT != 8. lol

**Pull request merged**